### DISCUSSION TITLE
Aws S3 Product files storage error

### DISCUSSION THREAD

**Discussion by Andrea-Tecnotrade** [OFFLINE-GmbH/oc-mall-plugin#1108] [Q&A]
Hello,
i choose s3  as storage, but i can't upload nor read from products.
I looked for an event i can attch but i didn't find.
I thnik that, as you know, that the problem is not possible to create the directory tree on aws as file has been uploaded

**Comment by SamBrishes** [comment]
Hello,

I'm not really aware of what an AWS S3 really is (looks like a simple cloud storage space) but, however, do you receive any error message or log file from AWS or October / Laravel (the Laravel Log can be found at `/storage/logs/system.log`).

I guess you're using your AWS S3 OctoberCMS / Laravel-wide as your global storage engine. Thus, can you please check if you can upload images using the OctoberCMS backend / Media browser?

Sincerely,\
Sam.

**Reply by Andrea-Tecnotrade** [reply]
Hello Sam for my delay in answering,
yes the problem is not with media folder, i have that problem with upladed with the offline mall plugin because i can't "intercept" the uploaded file events (both for single products and for variants) . As you know Offline.Mall create image sets related to product and variant models i shoul get the file once is uploaded from backend, the move the file to known folder and save it with it's original name to s3 storage. While i am able to do in other simpler plugin, i can't get the uploaded file, move it to s3, and correctly store path in product image sets. I didn't find an event to "capture" the uploaded file i mall product. 
Thank you for your collaboration. 
Sincerely
Andrea
